### PR TITLE
Test for inline blocks with children with orthogonal writing modes and margins

### PR DESCRIPTION
**Pull request by svillar**
This was created after [this WebKit bug](https://bugs.webkit.org/show_bug.cgi?id=233562).

In order to compute the shrink to fit inline size of an inline block we must compute first the inline sizes of our children. This includes margins as well. The bug in WebKit was that it was always using the inline (from the child POV) margins of its children. That works if the children are parallel to the parent, but if any of them is orthogonal then we should use the margins in the block direction of the child (which correspond to the inline direction of the parent).

### CODE REVIEW

File: css/css-writing-modes/inline-box-orthogonal-child-with-margins-ref.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  border-style: dotted;
+}
+.container > div:nth-child(1) { background: yellow; }
+.container > div:nth-child(2) { background: purple; }
+
+</style>
+
+<div class="container">
+  <div></div>
+  <div style="writing-mode: vertical-lr;"></div>
```
Comment (mrego) [inline css/css-writing-modes/inline-box-orthogonal-child-with-margins-ref.html:27]: Can we avoid using orthogonal writing modes in the reference?

File: css/css-writing-modes/inline-box-orthogonal-child-with-margins.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  margin:       10px 20px 30px 40px;
+  border-width: 20px 30px 40px 50px;
+  padding:      30px 40px 50px 60px;
+  border-style: dotted;
+}
+.container > div:nth-child(1) { background: yellow; }
+.container > div:nth-child(2) { background: purple; }
+
+</style>
+
```
Comment (mrego) [inline css/css-writing-modes/inline-box-orthogonal-child-with-margins.html:24]: It'd be nice to have the usual message, "the test passes if you see"; or as a minimum a "meta assert" with some explanation.

File: css/css-writing-modes/inline-box-orthogonal-child-with-margins-ref.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<link rel="author" title="Sergio Villar Senin" href="mailto:svillar@igalia.com"/>
+<link rel="help" href="https://drafts.csswg.org/css-writing-modes-3/#block-flow"
```
Comment (mrego) [inline css/css-writing-modes/inline-box-orthogonal-child-with-margins-ref.html:3]: You don't need this in the reference file.

File: css/css-writing-modes/inline-box-orthogonal-child-with-margins.html (hunk 2 above)
Comment (svillar) [reply css/css-writing-modes/inline-box-orthogonal-child-with-margins.html:24]: ok

File: css/css-writing-modes/inline-box-orthogonal-child-with-margins-ref.html (hunk 3 above)
Comment (svillar) [reply css/css-writing-modes/inline-box-orthogonal-child-with-margins-ref.html:3]: right

**Review by mrego — APPROVED** [review]

**Pull request merged**